skopt/tree_opt.py

```diff
@@ -51,11 +54,27 @@ def _tree_minimize(func, dimensions, base_estimator, maxiter,
         # has zero gradient over large distances. As a result we can not
         # use gradient based optimisers like BFGS, so using random sampling
         # for the moment.
-        X = space.transform(space.rvs(n_samples=n_points,
-                                      random_state=rng))
-        values = -gaussian_ei(X, rgr, np.min(yi))
-        next_x = X[np.argmin(values)]
-
+        curr_min = np.min(yi)
+        if acq_min == "sampling":
+            X = space.transform(space.rvs(n_samples=n_points,
+                                          random_state=rng))
+            values = -gaussian_ei(X, rgr, curr_min)
+            next_x = X[np.argmin(values)]
+        elif acq_min == "paramils":
+            for dimension in space.dimensions:
+                if isinstance(dimension, Categorical):
```

Comment:
It is not straightforward to support the categorical case, without changing the space API to optionally not transform some of the dimensions. I would prefer handling this later.